tests/unit/s2n_client_key_share_extension_pq_test.c

```diff
@@ -0,0 +1,1069 @@
+                /* The key shares in this extension are fake - that's OK, the server should ignore the
+                 * KEM group ID and skip the share. */
+                EXPECT_SUCCESS(s2n_stuffer_alloc_ro_from_hex_string(&key_share_extension,
+                        /* Shares size: 12 bytes */
+                        "000C"
+                        /* IANA ID for secp256r1_sikep434r2 */
+                        "2F1F"
+                        /* KEM group share size: 8 bytes */
+                        "0008"
+                        /* ECC share size: 2 bytes */
+                        "0002"
+                        /* Fake ECC share */
+                        "FFFF"
+                        /* PQ share size: 2 bytes */
+                        "0002"
+                        /* Fake PQ share */
+                        "FFFF"
+                ));
+
+                EXPECT_SUCCESS(s2n_client_key_share_extension.recv(server_conn, &key_share_extension));
+
+                /* .recv should have read all data */
+                EXPECT_EQUAL(s2n_stuffer_data_available(&key_share_extension), 0);
+
+                /* Server should not have accepted any key shares */
+                const struct s2n_ecc_preferences *ecc_pref = NULL;
+                EXPECT_SUCCESS(s2n_connection_get_ecc_preferences(server_conn, &ecc_pref));
+                EXPECT_NOT_NULL(ecc_pref);
+
+                for (size_t ec_index = 0; ec_index < ecc_pref->count; ec_index++) {
+                    struct s2n_ecc_evp_params *received_params = &server_conn->secure.client_ecc_evp_params[ec_index];
+                    EXPECT_NULL(received_params->negotiated_curve);
+                    EXPECT_NULL(received_params->evp_pkey);
+                }
+
+                const struct s2n_kem_preferences *server_kem_pref = NULL;
+                EXPECT_SUCCESS(s2n_connection_get_kem_preferences(server_conn, &server_kem_pref));
+                EXPECT_NOT_NULL(server_kem_pref);
+
+                for (size_t pq_index = 0; pq_index < server_kem_pref->tls13_kem_group_count; pq_index++) {
+                    struct s2n_kem_group_params *received_params = &server_conn->secure.client_kem_group_params[pq_index];
+                    EXPECT_NULL(received_params->kem_group);
+                    EXPECT_NULL(received_params->ecc_params.negotiated_curve);
+                    EXPECT_NULL(received_params->ecc_params.evp_pkey);
+                    EXPECT_NULL(received_params->kem_params.kem);
+                    EXPECT_NULL(received_params->kem_params.public_key.data);
+                    EXPECT_EQUAL(received_params->kem_params.public_key.size, 0);
+                    EXPECT_EQUAL(received_params->kem_params.public_key.allocated, 0);
+                }
+
+                /* Server should have indicated HRR */
+                EXPECT_TRUE(s2n_is_hello_retry_handshake(server_conn));
+
+                EXPECT_SUCCESS(s2n_connection_free(server_conn));
+            } else {
```

Comment:
The previous if statement is far enough away that I think you should probably just do a fresh `if(!s2n_is_in_fips_mode)` check here. It'll also make it clear that this is a separate, independent test.